app/api/tests/test_api.py

```diff
@@ -715,95 +715,144 @@ def test_disallows_non_annotation_approver_to_approve_and_disapprove_labels(self
+        self.assertEqual(len(response.data), 3)
+
+        self.client.login(username=self.another_project_member_name,
+                        password=self.another_project_member_pass)
+        response = self.client.get(self.url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 3)
+
+    def test_does_not_return_comments_to_non_project_member(self):
+        self.client.login(username=self.non_project_member_name,
+                        password=self.non_project_member_pass)
+        response = self.client.get(self.url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_does_not_allow_deletion_by_non_project_member(self):
+        self.client.login(username=self.non_project_member_name,
+                        password=self.non_project_member_pass)
+        response = self.client.delete('{}/{}'.format(self.url, self.comment.id), format='json')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_does_not_allow_deletion_of_non_owned_comment(self):
+        self.client.login(username=self.another_project_member_name,
+                        password=self.another_project_member_pass)
+        response = self.client.delete('{}/{}'.format(self.url, self.comment.id), format='json')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_create_update_delete_comment(self):
+        self.client.login(username=self.project_member_name,
+                        password=self.project_member_pass)
+        response = self.client.post(self.url, format='json', data={'text': 'comment'})
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data['user'], self.project_member.id)
+        self.assertEqual(response.data['text'], 'comment')
+        url = '{}/{}'.format(self.url, response.data['id'])
+
+        # check if all comments are fetched
+        response = self.client.get(self.url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 4)
+
+        # update comment
+        response = self.client.patch(url, format='json', data={'text': 'new comment'})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['text'], 'new comment')
+
+        # delete comment
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        response = self.client.get(self.url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 3)
+
+    @classmethod
+    def doCleanups(cls):
+        remove_all_role_mappings()
+
+
+class TestCommentProjectListAPI(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
```

Comment:
the only difference is `cls.url`. please merge both classes and define the second url in `cls.url_project`or something